### DISCUSSION TITLE
How to keep correctly placed pieces?

### DISCUSSION THREAD

**Discussion by sebastiaandegeus** [flbulgarelli/headbreaker#50] [Q&A] [answered]
Hi, 

First of all, great library. I have a couple questions that I could not figure out from the docs:

- How can I prevent correctly placed pieces to be disconnected?
- How can I drag correctly placed pieces as a group? (I guess if they cannot be disconnected this would also be solved)
- How can I prevent the connecting of wrong pieces? Right now they seem to _stick_

Thank you.

**Comment by flbulgarelli** [comment, marked as answer]
Hi @sebastiaandegeus :wave: 

Thanks for your comments. I think your first two questions are going to get answered by yesterday's deploy - https://github.com/flbulgarelli/headbreaker/releases/tag/v2.0.0. Please see https://github.com/flbulgarelli/headbreaker/discussions/45

The third question is more tricky, and the short answer is that there is nothing built-in that prevents "wrong" connections. But you can deal with this problem by using validators - see https://github.com/flbulgarelli/headbreaker/discussions/35 - to report to the user that the piece is wrong. Alternatively, you could use `onConnect` event in `Puzzle` o `Piece` to force disconnection under such circumstances. 

Hope it helps!
